### PR TITLE
Add personal_sign & personal_ecRecover

### PR DESCRIPTION
**Pull request by ellis2323**
Hi,

I have added these two standard methods:

 * personal_sign
 * personal_ecRecover

and i have added a unit test which check the personal_sign and personal_ecRecover works well together.

As the EIP-712 is not yet widely adopted, many use these two methods.

Best regards,

### CODE REVIEW

File: lib/statemanager.js [hunk 1]
```diff
@@ -449,6 +449,20 @@ StateManager.prototype.sign = function(address, dataToSign) {
...
+
+  if (!account) {
+    throw new Error("cannot sign data; no private key");
+  }
+
+  return sigUtil.personalSign(account.secretKey, { data: dataToSign });
+};
+
+StateManager.prototype.personal_ecRecover = function(dataThatWasSigned, signature) {
+  return sigUtil.recoverPersonalSignature({ data: dataThatWasSigned, sig: signature });
```
Comment (princesinha19) [inline lib/statemanager.js]: I think it's not good to use the abbreviation. I would rename **sig** to **signature**.

File: lib/subproviders/geth_api_double.js [hunk 2]
```diff
@@ -511,6 +511,32 @@ GethApiDouble.prototype.personal_sendTransaction = function(txData, password, ca
...
+  return callback(error, result);
+};
+
+GethApiDouble.prototype.personal_ecRecover = function(dataThatWasSigned, signature, callback) {
+  var result;
+  var error;
+
+  try {
+    result = this.state.personal_ecRecover(dataThatWasSigned, signature);
+  } catch (e) {
```
Comment (princesinha19) [inline lib/subproviders/geth_api_double.js]: same here. I would replace **e** with **error**.

File: lib/subproviders/geth_api_double.js [hunk 3]
```diff
@@ -511,6 +511,32 @@ GethApiDouble.prototype.personal_sendTransaction = function(txData, password, ca
   });
 };
 
+GethApiDouble.prototype.personal_sign = function(dataToSign, address, password, callback) {
+  var result;
+  var error;
+
+  try {
+    result = this.state.personal_sign(address, dataToSign);
+  } catch (e) {
```
Comment (princesinha19) [inline lib/subproviders/geth_api_double.js]: same here. I can remove abbreviation.

File: lib/statemanager.js (hunk 1 above)
Comment (ellis2323) [reply lib/statemanager.js]: ok

**Comment by davidmurdoch** [comment]
Closing this PR since this is based off of the old v6 architecture. We're tracking the implementation in v7 in https://github.com/trufflesuite/ganache/issues/995

**Pull request closed without merging**